pkg/controllers/pod_controller.go

```diff
@@ -231,3 +271,11 @@ func usingExclusivePlacement(pod *corev1.Pod) bool {
 func isLeaderPod(pod *corev1.Pod) bool {
 	return pod.Annotations[batchv1.JobCompletionIndexAnnotation] == "0"
 }
+
+func leaderPodName(pod *corev1.Pod) string {
+	parts := strings.Split(pod.Name, "-")
+	n := len(parts)
+	parts = parts[:n-1] // remove pod index
+	parts = append(parts, "0")
```

Comment:
This is not correct because the pod have a random suffix added to it; we need to look up the leader pod using the index label

Reply:
Since index isn't added as a label in version < 1.28, I opted to get pods with the same job-key and then return the one with completion index annotation of 0.